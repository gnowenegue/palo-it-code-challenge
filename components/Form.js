import React, { useContext } from 'react';

import Router from 'next/router';

import AppContext from '../contexts/AppContext';

const Form = () => {
  const { setUser } = useContext(AppContext);

  const handleSubmit = e => {
    e.preventDefault();

    const form = document.userForm;
    const formData = new FormData(form);

    setUser({
      firstName: formData.get('firstName'),
      lastName: formData.get('lastName'),
      email: formData.get('email'),
    });

    Router.push('/illnesses');
  };

  return (
    <form
      name="userForm"
      id="userForm"
      onSubmit={handleSubmit}
      className="col s8 offset-s2"
    >
      <h5>Who are you?</h5>
      <div className="row">
        <div className="input-field col s6">
          <input
            name="firstName"
            id="firstName"
            type="text"
            className="validate"
            required
          />
          <label htmlFor="firstName">First Name</label>
          <span
            className="helper-text"
            data-error="Please enter your first name."
          />
        </div>
        <div className="input-field col s6">
          <input
            name="lastName"
            id="lastName"
            type="text"
            className="validate"
            required
          />
          <label htmlFor="lastName">Last Name</label>
          <span
            className="helper-text"
            data-error="Please enter your last name."
          />
        </div>
        <div className="input-field col s12">
          <input
            name="email"
            id="email"
            type="email"
            className="validate"
            required
          />
          <label htmlFor="email">Email</label>
          <span
            className="helper-text"
            data-error="Please enter a valid email."
          />
        </div>
        <div className="col s12 center-align">
          <button className="btn waves-effect waves-light" type="submit">
            Submit
            <i className="material-icons right">send</i>
          </button>
        </div>
      </div>
      <style jsx>
        {`
          .btn {
            margin-top: 2rem;
          }
        `}
      </style>
    </form>
  );
};

export default Form;
