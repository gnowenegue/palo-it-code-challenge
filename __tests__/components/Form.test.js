import React from 'react';
import renderer from 'react-test-renderer';
// import { mount, shallow, render } from 'enzyme';

import AppContext from '../../contexts/AppContext';
import Form from '../../components/Form';

describe('<Form />', () => {
  test('Snapshot test', () => {
    const render = renderer.create(
      <AppContext.Provider value={{ setUser: () => {} }}>
        <Form />
      </AppContext.Provider>
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
