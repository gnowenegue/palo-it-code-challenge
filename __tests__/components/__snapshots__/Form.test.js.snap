// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> Snapshot test 1`] = `
<form
  className="jsx-3607409965 col s8 offset-s2"
  id="userForm"
  name="userForm"
  onSubmit={[Function]}
>
  <h5
    className="jsx-3607409965"
  >
    Who are you?
  </h5>
  <div
    className="jsx-3607409965 row"
  >
    <div
      className="jsx-3607409965 input-field col s6"
    >
      <input
        className="jsx-3607409965 validate"
        id="firstName"
        name="firstName"
        required={true}
        type="text"
      />
      <label
        className="jsx-3607409965"
        htmlFor="firstName"
      >
        First Name
      </label>
      <span
        className="jsx-3607409965 helper-text"
        data-error="Please enter your first name."
      />
    </div>
    <div
      className="jsx-3607409965 input-field col s6"
    >
      <input
        className="jsx-3607409965 validate"
        id="lastName"
        name="lastName"
        required={true}
        type="text"
      />
      <label
        className="jsx-3607409965"
        htmlFor="lastName"
      >
        Last Name
      </label>
      <span
        className="jsx-3607409965 helper-text"
        data-error="Please enter your last name."
      />
    </div>
    <div
      className="jsx-3607409965 input-field col s12"
    >
      <input
        className="jsx-3607409965 validate"
        id="email"
        name="email"
        required={true}
        type="email"
      />
      <label
        className="jsx-3607409965"
        htmlFor="email"
      >
        Email
      </label>
      <span
        className="jsx-3607409965 helper-text"
        data-error="Please enter a valid email."
      />
    </div>
    <div
      className="jsx-3607409965 col s12 center-align"
    >
      <button
        className="jsx-3607409965 btn waves-effect waves-light"
        type="submit"
      >
        Submit
        <i
          className="jsx-3607409965 material-icons right"
        >
          send
        </i>
      </button>
    </div>
  </div>
</form>
`;
